### ISSUE TITLE
libzvbi configure fails on aarch64

### ISSUE THREAD

**Issue by paolo-sz** [xbmc/inputstream.ffmpegdirect#68]
The --build option is not passed to libzvbi configure through CMakeList.txt and config.guess (a little old) is not able to detect correctly the build triplet in case of aarch64-*-*.
This breaks the build of libzvbi dependency breaking consequently the build of inputstream.ffmpegdirect addon.

**Comment by phunkyfish** [comment]
If you have a fix for this please create a PR ;)

**Comment by paolo-sz** [comment]
Well, I thought to two options:
1. add the --build option to CMakeList.txt
2. update config.guess

Even if option 2. is "simpler" I would prefer option 1. (more robust). What do you prefer?
But, in case of option 1., I need some hints (build flow is a little bit hard to follow...) on how to generate the value to assign to --build (I mean which are, if they exist, the variables that contains the three values of the triplet?).

**Comment by phunkyfish** [comment]
What do you expect the triplet to be?

I think what we could do is just supply this argument for aarch64 so we don't impact any other platforms.

Is this for any aarch64 platform or only certain ones?

**Comment by paolo-sz** [comment]
Mine triplet output from <code>gcc -dumpmachine</code> is aarch64-linux-gnu and config.guess is not able to manage any triplet where the first field is aarch64 so, potentially, other platforms could be impacted

**Comment by paolo-sz** [comment]
Unfortunately things are not that simple as I was thinking (strange... it never happens... ;-))
I made a couple of tries and I discovered not only config.guess but also configure does not support aarch64 architecture... arm64 (sometime an alias) is not supported as well. It seems the whole autotool stuff is "old" and should be updated by an autoreconf but then some of the patches may be overwritten... I will continue to look into this issue as soon as I have some spare time (hope in a couple of days).

**Comment by phunkyfish** [comment]
zvbi has not been updated for many years. But vlc also build it so it must be possible.

For instance it is built for iOS which uses aarch64 so it must work somehow.

No rush

**Comment by wsnipex** [comment]
without looking into this: have you tried running autoreconf -rvi to generate a new configure script?

**Comment by paolo-sz** [comment]
@wsnipex it is exactly what I would like to try as soon I have 5 minutes in front of my HW.
My doubts are about the patches that are applied before the configure command (that should became a <code>autoreconf && configure</code>). But, at the first look, all the patches are applied also to .am and .in files which means they are not lost by an autoreconf... but I'm pretty sure things cannot be so simple ;-)...

**Comment by phunkyfish** [comment]
But sometimes (not often) the simple solution works!

**Comment by paolo-sz** [comment]
It tooks us 5 minutes to think to the fix (autoreconf) and 4 hours in order to integrate in this diabolic union of evil environments (kodi build system, cmake and autotools). ;-)

BTW autoreconf requires autopoint (I was not able to understand if it is part of autotool suite) to be installed...
